src/plugins/discord/mirror.test.js

```diff
@@ -0,0 +1,367 @@
+      const guildId = "0";
+      const db = new Database(":memory:");
+      const sqliteMirror = new SqliteMirror(db, guildId);
+
+      const members = jest
+        .fn()
+        .mockImplementationOnce(() => {
+          return Promise.resolve({
+            results: [testMember("1"), testMember("2")],
+            pageInfo: {hasNextPage: true, endCursor: "2"},
+          });
+        })
+        .mockImplementationOnce(() => {
+          return Promise.resolve({
+            results: [testMember("3")],
+            pageInfo: {hasNextPage: false, endCursor: "3"},
+          });
+        });
+
+      const fetcher = new MockFetcher({
+        channels: emptyChannels,
+        members,
+        messages: emptyMessages,
+        reactions: emptyReactions,
+      });
+      const mirror = new Mirror(guildId, sqliteMirror, fetcher);
+      await mirror.update();
+      expect(sqliteMirror.members()).toEqual([
+        testMember("1"),
+        testMember("2"),
+        testMember("3"),
+      ]);
+    });
+    it("fetches messages", async () => {
+      const guildId = "1";
+      const channelId = "2";
+      const authorId = "3";
+      const db = new Database(":memory:");
+      const sqliteMirror = new SqliteMirror(db, guildId);
+
+      const testMessage = (id: Snowflake): Model.Message => ({
+        id: id,
+        channelId: channelId,
+        authorId: authorId,
+        timestampMs: Date.parse("2020-03-03T23:35:10.615000+00:00"),
+        content: "Test message content",
+        reactionEmoji: [],
+        nonUserAuthor: false,
+        mentions: [],
+      });
+
+      const channels = jest
+        .fn()
+        .mockImplementation(() => Promise.resolve([testChannel(channelId)]));
+
+      const members = jest.fn().mockImplementationOnce(() => {
+        return Promise.resolve({
+          results: [testMember(authorId)],
+          pageInfo: {hasNextPage: false, endCursor: authorId},
+        });
+      });
+
+      const messages = jest
+        .fn()
+        .mockImplementationOnce(() => {
+          return Promise.resolve({
+            results: [testMessage("1"), testMessage("2")],
+            pageInfo: {hasNextPage: true, endCursor: "2"},
+          });
+        })
+        .mockImplementationOnce(() => {
+          return Promise.resolve({
+            results: [testMessage("3")],
+            pageInfo: {hasNextPage: false, endCursor: "3"},
+          });
+        });
+
+      const fetcher = new MockFetcher({
+        channels,
+        members,
+        messages,
+        reactions: emptyReactions,
+      });
+
+      const mirror = new Mirror(guildId, sqliteMirror, fetcher);
+      await mirror.update();
+      expect(sqliteMirror.messages(channelId)).toEqual([
+        testMessage("1"),
+        testMessage("2"),
+        testMessage("3"),
+      ]);
+      expect(messages.mock.calls).toEqual([
+        [channelId, "0"],
+        [channelId, "2"],
+      ]);
+    });
+
+    it("fetches reactions", async () => {
```

Comment:
I'm not super happy with the amount of state we need to build up to run this test. An option I considered was testing that the `_fetchReactions` method was called with the correct parameters and then testing `_fetchReactions` in isolation. However, that felt like testing the implementation versus the behavior. Note, I'm not crystal clear on that distinction/testing methodology.